src/Umbraco.Core/IO/UmbracoMediaFile.cs

```diff
@@ -86,7 +86,7 @@ public static UmbracoMediaFile Save(HttpPostedFileBase file)
         private void Initialize()
         {
             Filename = _fs.GetFileName(Path);
-            Extension = _fs.GetExtension(Path) != null

```

Comment:
looks good, only thing is that our code rules require to use `string.IsNullOrEmpty(_fs.GetExtension(Path)) == false` instead of `!string.IsNullOrEmpty(_fs.GetExtension(Path))`